core/services/relay/evm/relayer_extender.go

```diff
@@ -64,12 +68,13 @@ func NewLegacyChains(ctx context.Context, opts legacyevm.ChainRelayOpts) (result
 		}
 
 		privOpts.Logger.Infow(fmt.Sprintf("Loading chain %s", cid), "evmChainID", cid)
-		chain, err2 := legacyevm.NewTOMLChain(ctx, enabled[i], privOpts)
+		chain, err2 := legacyevm.NewTOMLChain(ctx, enabled[i], privOpts, chainIDToClientMap)
```

Comment:
nice, yeah looking at the implementation prior to this with the func compared to just passing in the map .. this way feels cleaner